helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -602,4 +590,28 @@ private void removeContextsAndPreviousAssignment(String workflow, Set<String> jo
     }
     cache.removeContext(workflow);
   }
+
+  private long calculateStartTime(WorkflowContext workflowCtx, String job, JobConfig jobConfig) {
```

Comment:
I think computeStartTimeForJob or something like that is more intuitive and specific. What do you think?